spec/lib/identity_tijuana_member_sync_spec.rb

```diff
@@ -0,0 +1,325 @@
+RSpec.describe IdentityTijuana::MemberSync do
+  describe '#get_id_change_date' do
+    before(:each) do
+      @email_sub = Subscription::EMAIL_SUBSCRIPTION
+      @calling_sub = Subscription::CALLING_SUBSCRIPTION
+      @sms_sub = Subscription::SMS_SUBSCRIPTION
+
+      allow(Settings).to receive_message_chain(
+        "options.default_phone_country_code"
+      ) { '61' }
+      allow(Settings).to receive_message_chain(
+        "options.default_mobile_phone_national_destination_code"
+      ) { 4 }
+      allow(Settings).to receive_message_chain(
+        "tijuana.email_subscription_id"
+      ) { @email_sub.id }
+      allow(Settings).to receive_message_chain(
+        "tijuana.calling_subscription_id"
+      ) { @calling_sub.id }
+      allow(Settings).to receive_message_chain(
+        "tijuana.sms_subscription_id"
+      ) { @sms_sub.id }
+      allow(Settings).to receive_message_chain(
+        "geography.postcode_dash"
+      ) { false }
+      allow(Settings).to receive_message_chain(
+        "geography.area_lookup.track_area_probabilities"
+      ) { false }
+    end
+
+    context 'with multiple audit logs entries' do
+      it 'correctly identifies the change date for name' do
+        member = FactoryBot.create(:member)
+        member.update!(first_name: 'NewName')
+
+        sleep(0.2)
```

Comment:
Prefer using the [time helpers](https://guides.rubyonrails.org/testing.html#testing-time-dependent-code) instead of sleeping since it gives you more control, and doesn't slow down the unit tests.

Reply:
/nice – changed to `travel_to`